added_values.go

```diff
@@ -0,0 +1,608 @@
+	return strconv.FormatInt(int64(i.ToInt()), 10)
+}
+
+func (i valueNumber) string() unistring.String {
+	return unistring.String(strconv.FormatInt(int64(i.ToInt()), 10))
+}
+
+func (i valueNumber) ToFloat() float64 {
+	return float64(i.ToInt64())
+}
+
+func (i valueNumber) ToBoolean() bool {
+	return i.val != 0
+}
+
+func (i valueNumber) ToObject(r *Runtime) *Object {
+	return r.newPrimitiveObject(i, r.global.NumberPrototype, classNumber)
+}
+
+func (i valueNumber) ToNumber() Value {
+	return i
+}
+
+func (i valueNumber) SameAs(other Value) bool {
+	if otherInt, ok := other.assertInt(); ok {
+		return i.ToInt() == otherInt
+	}
+	if otherFloat, ok := other.assertFloat(); ok {
+		return i.ToFloat() == otherFloat
+	}
+	return false
+}
+
+func (i valueNumber) Equals(other Value) bool {
+	if o, ok := other.assertInt(); ok {
+		return i.ToInt() == o
+	}
+	if o, ok := other.assertFloat(); ok {
+		return float64(i.ToInt()) == o
+	}
+	if o, ok := other.assertString(); ok {
+		return o.ToNumber().Equals(i)
+	}
+	if o, ok := other.(valueBool); ok {
+		return int(i.ToInt()) == o.ToInt()
+	}
+	if o, ok := other.(*Object); ok {
+		return i.Equals(o.self.toPrimitiveNumber())
+	}
+	return false
+}
+
+func (i valueNumber) StrictEquals(other Value) bool {
+	if otherInt, ok := other.assertInt(); ok {
+		return int(i.ToInt()) == otherInt
+	} else if otherFloat, ok := other.assertFloat(); ok {
+		return float64(i.ToInt()) == otherFloat
+	}
+	return false
+}
+
+func (i valueNumber) assertInt() (int, bool) {
+	return i.ToInt(), true
+}
+func (i valueNumber) assertInt32() (int32, bool) {
+	return i.ToInt32(), true
+}
+func (i valueNumber) assertUInt32() (uint32, bool) {
+	return i.ToUInt32(), true
+}
+func (i valueNumber) assertInt64() (int64, bool) {
+	return i.ToInt64(), true
+}
+
+func (i valueNumber) assertFloat() (float64, bool) {
+	return i.ToFloat(), false
+}
+
+func (i valueNumber) assertString() (valueString, bool) {
+	return nil, false
+}
+
+func (i valueNumber) baseObject(r *Runtime) *Object {
+	return r.global.NumberPrototype
+}
+
+func (i valueNumber) Export() interface{} {
+	return i.val
+}
+
+func (i valueNumber) ExportType() reflect.Type {
+	return i._type
+}
+
+func (i valueUInt32) MemUsage(ctx *MemUsageContext) (uint64, error) {
+	return NumberSize, nil
+}
+
+func (i valueUInt32) ToInt() int {
+	return int(i)
+}
+
+func (i valueUInt32) ToInt32() int32 {
+	return int32(i)
+}
+func (i valueUInt32) ToUInt32() uint32 {
+	return uint32(i)
+}
+
+func (i valueUInt32) ToInt64() int64 {
+	return int64(i)
+}
+func (i valueUInt32) ToInteger() int64 {
+	return int64(i)
+}
+
+func (i valueUInt32) ToString() Value {
+	return asciiString(i.String())
+}
+func (i valueUInt32) toString() valueString {
+	return asciiString(i.String())
+}
+func (i valueUInt32) String() string {
+	return strconv.FormatInt(int64(i.ToInt()), 10)
+}
+
+func (i valueUInt32) string() unistring.String {
+	return unistring.String(strconv.FormatInt(int64(i.ToInt()), 10))
```

Comment:
[nit] `return unistring.String(i.String())`? This applies to the other `string()` functions below as well.